src/main/java/com/google/api/codegen/transformer/ModelTypeTable.java

```diff
@@ -161,8 +161,8 @@ public String getAndSaveNicknameForContainer(
    * If the given type is not implicitly imported, the add it to the import set, then return the
    * zero value for that type.
    */
-  public String getZeroValueAndSaveNicknameFor(TypeRef type) {
-    return typeNameConverter.getZeroValue(type).getValueAndSaveTypeNicknameIn(typeTable);
+  public String getZeroValueAndSaveNicknameFor(TypeRef type, ZeroValuePurpose purpose) {
```

Comment:
I think I would prefer to introduce a new function instead of an enum for supporting the distinction you want. Something like `getParamZeroValue`.

For my understanding: why is this distinction necessary?